helix-rest/src/main/java/org/apache/helix/rest/clusterMaintenanceService/MaintenanceManagementService.java

```diff
@@ -67,10 +67,17 @@
 
 
 public class MaintenanceManagementService {
+  private static class MaintenanceExecutorService {
+    static final int totalInstance = 1000;
+    static final int faultZones = 20;
+    public static ExecutorService getExecutorPool() {
+      return Executors.newFixedThreadPool(totalInstance / faultZones);
+    }
+  }
 
   private static final Logger LOG = LoggerFactory.getLogger(MaintenanceManagementService.class);
   private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
-  private static final ExecutorService POOL = Executors.newCachedThreadPool();
+  private static final ExecutorService POOL = MaintenanceExecutorService.getExecutorPool();
```

Comment:
Please add comments which address how this POOL is used. Is it 1 thread per instance that is having checks run on it?

In the place where the pool is used, explain what happens if all threads are being used. Explain how we are queuing the next checks to run.